tests/integration/test_interface.py

```diff
@@ -6,77 +6,96 @@
+    dataset: Dataset, noising_function: Callable, use_sample_data: bool, tmpdir
 ):
     """Tests that noised datasets are generated and as expected. The 'use_sample_data'
     parameter determines whether to use the sample data (if True) or
     a non-default root directory with multiple datasets to compile (if False)
     """
     if noising_function == "todo":
-        pytest.skip(reason=f"TODO: implement dataset {data_dir_name}")
+        pytest.skip(reason=f"TODO: implement dataset {dataset}")
 
-    sample_data_path = list(
-        (paths.SAMPLE_DATA_ROOT / data_dir_name).glob(f"{data_dir_name}*")
-    )[0]
+    sample_data_path = list((paths.SAMPLE_DATA_ROOT / dataset.name).glob(f"{dataset.name}*"))[
+        0
+    ]
 
     data = _load_data(sample_data_path)
 
     # Configure if default (sample data) is used or a different root directory
     if use_sample_data:
         source = None  # will default to using sample data
     else:
-        source = _generate_non_default_data_root(
-            data_dir_name, tmpdir, sample_data_path, data
-        )
+        source = _generate_non_default_data_root(dataset.name, tmpdir, sample_data_path, data)
 
-    noised_data = noising_function(seed=0, source=source)
-    noised_data_same_seed = noising_function(seed=0, source=source)
-    noised_data_different_seed = noising_function(seed=1, source=source)
+    noised_data = noising_function(seed=0, source=source, year=None)
+    noised_data_same_seed = noising_function(seed=0, source=source, year=None)
+    noised_data_different_seed = noising_function(seed=1, source=source, year=None)
 
     assert not data.equals(noised_data)
     assert noised_data.equals(noised_data_same_seed)
     assert not noised_data.equals(noised_data_different_seed)
 
-    # Check each column's dtype
-    for col in noised_data.columns:
-        expected_dtype = [c.dtype_name for c in COLUMNS if c.name == col][0]
+    # Check each column
+    shared_idx = pd.Index(set(data.index).intersection(set(noised_data.index)))
+    check_original = _reformat_dates_for_noising(data.loc[shared_idx], dataset)
+    check_noised = noised_data.loc[shared_idx]
+
+    for col_name in noised_data.columns:
+        col = [c for c in COLUMNS if c.name == col_name][0]
+        # Check dtype is correct
+        expected_dtype = col.dtype_name
         if expected_dtype == np.dtype(str):
             # str dtype is 'object'
             expected_dtype = np.dtype(object)
-        assert noised_data[col].dtype == expected_dtype
+        assert noised_data[col_name].dtype == expected_dtype
+        # Check if noised
+        shared_not_missing_idx = shared_idx[
+            check_original[col_name].notna() & check_noised[col_name].notna()
+        ]
+        if col.noise_types:
+            assert (
+                check_original.loc[shared_not_missing_idx, col_name]
+                != check_noised.loc[shared_not_missing_idx, col_name]
+            ).mean() <= 0.04
```

Comment:
This will pass if original is equal to noised, which is not what we want to test ( 0 <= 0.04)

Reply:
omg, yeah ok. Hmm...

Reply:
Added assertion that there are `any()` differences